### PR TITLE
Feature/erc 1155 update

### PR DESCRIPTION
**Pull request by brolag**
## Description

This PR add the code for the ERC1155 contract required for the Cofi Collections.

## Type of Change

- [ ] Bugfix
- [x] Feature
- [ ] Refactor
- [ ] Chore
- [ ] Documentation
- [ ] Test

### CODE REVIEW

File: apps/snfoundry/contracts/src/cofi_collection.cairo [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+#[starknet::contract]
+mod CofiCollection {
+    use openzeppelin::access::accesscontrol::AccessControlComponent;
+    use openzeppelin::access::accesscontrol::DEFAULT_ADMIN_ROLE;
+    use openzeppelin::introspection::src5::SRC5Component;
+    use openzeppelin::security::pausable::PausableComponent;
+    use openzeppelin::token::erc1155::ERC1155Component;
+    use openzeppelin::upgrades::UpgradeableComponent;
+    use openzeppelin::upgrades::interface::IUpgradeable;
```
Comment (jsdanielh) [inline apps/snfoundry/contracts/src/cofi_collection.cairo:92]:
```suggestion
    use openzeppelin::{access::accesscontrol::{AccessControlComponent, DEFAULT_ADMIN_ROLE}, introspection::src5::SRC5Component, security::pausable::PausableComponent, token::erc1155::ERC1155Component, upgrades::{UpgradeableComponent, interface::IUpgradeable}};
```

File: apps/snfoundry/contracts/src/test/test_cofi_collection.cairo [hunk 2]
```diff
@@ -0,0 +1,418 @@
+use contracts::cofi_collection::ICofiCollectionDispatcher;
+use contracts::cofi_collection::ICofiCollectionDispatcherTrait;
+
+use openzeppelin::utils::serde::SerializedAppend;
+
```
Comment (jsdanielh) [inline apps/snfoundry/contracts/src/test/test_cofi_collection.cairo:5]:
```suggestion
```

File: apps/snfoundry/contracts/src/test/test_cofi_collection.cairo [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+use snforge_std::{declare, ContractClassTrait, DeclareResultTrait, start_cheat_caller_address};
+use starknet::ContractAddress;
+
+fn OWNER() -> ContractAddress {
```
Comment (jsdanielh) [inline apps/snfoundry/contracts/src/test/test_cofi_collection.cairo:9]:
```suggestion
const fn owner() -> ContractAddress {
```

File: apps/snfoundry/contracts/src/test/test_cofi_collection.cairo [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+    // Check balances after minting
+    let balance1 = cofi_collection.balance_of(receiver, token_id1);
+    let balance2 = cofi_collection.balance_of(receiver, token_id2);
+    assert(balance1 == mint_value1, 'Incorrect balance for token 1');
+    assert(balance2 == mint_value2, 'Incorrect balance for token 2');
+
+    // Test minting additional tokens
+    let additional_value1 = 5_u256;
+    let additional_value2 = 8_u256;
+    let additional_values: Array<u256> = array![additional_value1, additional_value2];
```
Comment (jsdanielh) [inline apps/snfoundry/contracts/src/test/test_cofi_collection.cairo:146]:
```suggestion
    let additional_values: Array<u256> = array![5_u256, 8_u256];
```

**Review by jsdanielh — COMMENTED** [review]
Some comments:

**Comment by jsdanielh** [comment]
One more suggestion: Use `scarb fmt` to use a standard code style

File: apps/snfoundry/contracts/src/test/test_cofi_collection.cairo [hunk 5]
```diff
@@ -0,0 +1,418 @@
+use contracts::cofi_collection::ICofiCollectionDispatcher;
```
Comment (jsdanielh) [inline apps/snfoundry/contracts/src/test/test_cofi_collection.cairo:1]: I'd recommend to use the test directory structure as mentioned in [the documentation](https://foundry-rs.github.io/starknet-foundry/testing/test-collection.html#the-tests-directory)

File: apps/snfoundry/contracts/Scarb.toml [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
 
 # See more keys and their definitions at https://docs.swmansion.com/scarb/docs/reference/manifest.html
 
 [dependencies]
-starknet = "2.6.4"
-openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", tag = "v0.14.0" }
-snforge_std = { git = "https://github.com/foundry-rs/starknet-foundry", tag = "v0.27.0" }
+starknet = ">=2.8.2"
+openzeppelin = { git = "https://github.com/OpenZeppelin/cairo-contracts.git", tag = "v0.17.0" }
+snforge_std = { git = "https://github.com/foundry-rs/starknet-foundry", tag = "v0.31.0" }
```
Comment (jsdanielh) [inline apps/snfoundry/contracts/Scarb.toml:11]: I'd also recommend to keep this list alphabetically sorted

**Review by jsdanielh — COMMENTED** [review]
A couple more :smile:

File: apps/snfoundry/contracts/src/test/test_cofi_collection.cairo (hunk 3 above)
Comment (brolag) [reply apps/snfoundry/contracts/src/test/test_cofi_collection.cairo:9]: This one didn't work for some reason.

File: apps/snfoundry/contracts/src/test/test_cofi_collection.cairo (hunk 4 above)
Comment (brolag) [reply apps/snfoundry/contracts/src/test/test_cofi_collection.cairo:146]: We need these variables later in the execution.

File: apps/snfoundry/contracts/src/test/test_cofi_collection.cairo (hunk 5 above)
Comment (brolag) [reply apps/snfoundry/contracts/src/test/test_cofi_collection.cairo:1]: For some reason it stops running the tests if I move to this structure. We'll kee it for the moment but lets discuss about it.

**Pull request merged**